public/app/plugins/datasource/loki/queryUtils.test.ts

```diff
@@ -201,6 +202,23 @@ describe('parseToArray', () => {
   });
 });
 
+describe('obfuscate', () => {
+  it('obfuscates on invalid query', () => {
+    expect(obfuscate('{job="grafana"')).toEqual('{@@@=@@@@@@@@@');
+  });
+  it('obfuscates on valid query', () => {
+    expect(
+      obfuscate('sum(sum_over_time({test="test"} |= `` | logfmt | __error__=`` | unwrap test | __error__=`` [10m]))')
+    ).toEqual('sum(sum_over_time({@@@@=@@@@@@} |= @@ | logfmt | @@@@@@@@@=@@ | unwrap @@@@ | @@@@@@@@@=@@ [10m]))');
+  });
+  it('obfuscates on arithmetic operation', () => {
+    expect(obfuscate('2 + 3')).toEqual('@ + @');
+  });
+  it('obfuscates a comment', () => {
+    expect(obfuscate('{job="grafana"} # test comment')).toEqual('{@@@=@@@@@@@@@} @@@@@@@@@@@@@@');
```

Comment:
I would just do a replacement once, like  `@=@` and that's it. And it gives us an extra level of obfuscation: not even string length.

Reply:
The problem is, you can't get other tokens from the original query if you change the length of those tokens. That's why I decided to replace every character. Feel free to improve it, but I was out of ideas how to handle it nicely.

Reply:
I'll give it a try.

Reply:
Done. Let me know what you think.